src/nnet3/nnet-general-component.cc

```diff
@@ -1374,5 +1374,105 @@ void ConstantComponent::UnVectorize(const VectorBase<BaseFloat> &params) {
 
 
 
+std::string DropoutMaskComponent::Info() const {
+  std::ostringstream stream;
+  stream << Type()
+         << ", output-dim=" << output_dim_
+         << ", dropout-proportion=" << dropout_proportion_
+         << ", exclusive=" << (exclusive_ ? "true" : "false");
+  return stream.str();
+}
+
+DropoutMaskComponent::DropoutMaskComponent():
+    output_dim_(-1), dropout_proportion_(0.5),
+    exclusive_(false) { }
```

Comment:
I think the 'exclusive' option can be removed (treat it as if it is always false).